FreeRTOS/FreeRTOS-Kernel
# Block `SIG_SUSPEND` in the main thread of the POSIX port so that `sigwait` works as expected

Description
-----------
The POSIX standard[^1] requires that `sigwait` should only be used with blocked signals, otherwise the behavior is undefined. When the POSIX port is used on macOS for example, the `sigwait` in `xPortStartScheduler` never returns because there is a handler set up to ignore the `SIG_SUSPEND` signal. This change removes the ignore handler and blocks the signal prior to `sigwait`ing for it in the main thread.

Test Steps
-----------
Build a basic example on a non-Linux POSIX system using the POSIX FreeRTOS port, and try to use `vTaskEndScheduler`.
Without these changes, the program does not exit on macOS. It continues to behave as expected on Linux.

```c
#include <FreeRTOS.h>
#include <task.h>

#include <limits.h>

static void fn(void *arg)
{
    vTaskEndScheduler();
}

int main(void)
{
    static StackType_t stack[PTHREAD_STACK_MIN];
    static StaticTask_t task;
    xTaskCreateStatic(fn, "task", PTHREAD_STACK_MIN, NULL, 1, stack, &task);
    vTaskStartScheduler();
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

[^1]: https://pubs.opengroup.org/onlinepubs/009604499/functions/sigwait.html
